src/optimize.jl

```diff
@@ -0,0 +1,195 @@
+
+"""
+    optimize(
+        problem,
+        objective   ::AbstractVariationalObjective,
+        restructure,
+        param_init  ::AbstractVector{<:Real},
+        max_iter    ::Int,
+        objargs...;
+        kwargs...
+    )              
+    optimize(
+        problem,
+        objective             ::AbstractVariationalObjective,
+        variational_dist_init,
+        max_iter              ::Int,
+        objargs...;
+        kwargs...
+    )              
+
+Optimize the variational objective `objective` targeting the problem `problem` by estimating (stochastic) gradients, where the variational approximation can be constructed by passing the variational parameters `param_init` or the initial variational approximation `variational_dist_init` to the function `restructure`.
```

Comment:
```suggestion
Optimize the variational objective `objective` targeting the problem `problem` by estimating (stochastic) gradients.

The variational approximation can be constructed by passing the variational parameters `param_init` or the initial variational approximation `variational_dist_init` to the function `restructure`.
```

Ideally,  the first line is a quick-to-read and not-too-long description of roughly what the method does since this is the first thing the user sees when they do `?f`.